### DISCUSSION TITLE
Caixa de lista de carga

### DISCUSSION THREAD

**Discussion by CarlosEX** [AvaloniaUI/Avalonia#5512] [Q&A] [answered]
Good evening, guys, I need your help. I'm trying to do something trivial and I'm not getting it. I've researched and only found content in WPF that I could not replicate in avalonia. I'm currently in this state here. What am I doing wrong to fill out this ListBox? Could someone provide me with an example of how I can do this padding and what good practices would be for this type of linking. If you have any repository for study, it will also be of great help.  Thank you very much.
![Captura de tela de 2021-02-18 19-17-10](https://user-images.githubusercontent.com/33513870/108429598-f6322980-721e-11eb-95b3-b32220c5e7da.png)

**Comment by jp2masa** [comment, marked as answer]
`ListBox.Items` is an enumerable. So for example you could do `this.DataContext = new Person[] { new Person() { ... } }` and then `<ListBox Items="{Binding}"` (this would bind the items to the `DataContext` value itself, which is the array).

**Reply by CarlosEX** [reply]
Veryooooo thank you!
Themy form opens, but is not yet loading the values.
I'll leave down here the code snippets so you can see if that's really right.

# CODE XAML
------------------
    <Grid>
        <ListBox Items="{Binding}" BorderBrush="Transparent" Grid.IsSharedSizeScope="True">
            <ListBox.ItemTemplate>
                <DataTemplate>
                    <Grid Margin="4">
                        <Grid.ColumnDefinitions>
                            <ColumnDefinition Width="Auto" SharedSizeGroup="Key" />
                            <ColumnDefinition Width="*" />
                        </Grid.ColumnDefinitions>
                        <TextBlock Background="Red" Text="{Binding FirstName}" FontWeight="Bold"  />
                        <TextBox Background="Blue" Grid.Column="1" Text="{Binding LastName}" />
                    </Grid>
                </DataTemplate>
            </ListBox.ItemTemplate>
        </ListBox>
    </Grid>

# CLASS PERSON
--------------------------------------
    public class Person
    {
            private string? _firistName;
            private string? _lastName;
            public string? FirstName { get => _firistName; set => _firistName = value; }
            public string? LastName { get => _lastName; set => _lastName = value; }
        }

--------------------------------------------------
# MainWindow.xaml.cs
----------------------------------------------

    public class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
            this.DataContext = new Person[]{new Person(){ FirstName="Carlos", LastName="Antonio" }};
            
#if DEBUG
            this.AttachDevTools();
#endif
            }
    }

**Reply by jp2masa** [reply]
No idea why that's not working, it looks right. Is the `Grid` directly on the `Window`?

**Reply by CarlosEX** [reply]
It hasn't worked yet, I removed the column settings and left only the links, but the data still does not load. I'm using the mvvm pattern. But I believe that something else is missing, I will continue studying to find out what it is, if you have any example or repository of how I can do this linking with listbox, please pass me. I'm new to this one with Avalonia still, and I'm getting a little. Still, thank you so much for the strength!

-------------------------

    <ListBox Items="{Binding}">
            <ListBox.ItemTemplate>
                <DataTemplate>
                    <Grid Margin="4">
                        <TextBlock Text="{Binding FirstName}" />
                        <TextBox Text="{Binding LastName}" />
                    </Grid>
                </DataTemplate>
            </ListBox.ItemTemplate>
    </ListBox>
    
------------------------

**Reply by jp2masa** [reply]
It works for me: [ListBoxBinding.zip](https://github.com/AvaloniaUI/Avalonia/files/6006983/ListBoxBinding.zip).

**Reply by CarlosEX** [reply]
True, I opened here and it worked for me too. You know what it was, is that in the model I spent earlier I used the templante *avalonia.mvvm* and I believe you are talking something to make it work. But if it's with the *avalonia.app*, it works. Well, now I know where to focus my strength. And veryoooooo thanks for the help!

**Reply by CarlosEX** [reply]
Just could not make it work in the mvvm standard. I'm going to study it this week to understand the architecture and where I'm supposed to move. If you or someone already know and can replicate this example, it will be great to give a clear in mind.

**Reply by CarlosEX** [reply]
Please help me with this simple question?
Based on the same example you sent me, I used this fill structure here.

----------//

## With literal values

    <ListBox VirtualizationMode="None">
                <ListBoxItem >
                    <StackPanel>
                        <TextBlock Text="Title"/>
                        <TextBlock Text="SubTitle" />
                    </StackPanel>
                </ListBoxItem>
            </ListBox>
-----------------------------------
## Error!
When I link, it gives me an exception

### Exception
***Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.***

      <ListBox Items="{Binding}" VirtualizationMode="None">
                <ListBoxItem >
                    <StackPanel>
                        <TextBlock Text="{Binding FirstName}"/>
                        <TextBlock Text="{Binding LastName}" />
                    </StackPanel>
                </ListBoxItem>
            </ListBox>

**Reply by jp2masa** [reply]
The problem with the MVVM template is probably that the main window `DataContext` gets overwritten on `App.axaml.cs`:

```cs
                desktop.MainWindow = new MainWindow
                {
                    DataContext = new MainWindowViewModel(),
                };
```

So if you want to set the `DataContext` in the `MainWindow` constructor, you have to remove that line.

**Reply by CarlosEX** [reply]
Thank you so much for your help and attention. I am studying the wpf documentation part to understand these elementary issues and not skate again.
